x-pack/plugins/apm/server/routes/correlations/queries/query_significant_correlations.ts

```diff
@@ -76,12 +77,51 @@ export const fetchSignificantCorrelations = async (
     )
   );
 
-  const latencyCorrelations: LatencyCorrelation[] = fulfilled.filter(
-    (d): d is LatencyCorrelation => d !== undefined
-  );
+  let fallbackResult: LatencyCorrelation | undefined;
+  const latencyCorrelations: LatencyCorrelation[] = [];
+
+  fulfilled.forEach((d: LatencyCorrelation | undefined) => {
+    if (d === undefined) return;
+    if (Array.isArray(d.histogram)) {
+      latencyCorrelations.push(d);
+    } else {
+      if (!fallbackResult) {
+        fallbackResult = d;
+      } else {
+        if (
+          d.ksTest > fallbackResult.ksTest &&
+          d.correlation > fallbackResult.correlation
+        ) {
+          fallbackResult = d;
+        }
+      }
+    }
+  });
```

Comment:
Any possibility of rewriting this to using `filter` and `map`? Immutable is often easier to reason about and less error prone.

Reply:
Updated here https://github.com/elastic/kibana/pull/126683/commits/9f57ca31de5d775db046dfc0b09fcf504fdba73f